src/pasta/mod.rs

```diff
@@ -25,3 +29,300 @@ impl crate::CurveAffineExt for EqAffine {
+        unsafe {
+            is_less_than(&inner, &std::mem::transmute_copy(&Fq::MODULUS))
+                .then(|| elt)
+                .ok_or_else(|| {
+                    std::io::Error::new(
+                        std::io::ErrorKind::InvalidData,
+                        "input number is not less than field modulus",
+                    )
+                })
+        }
+    }
+    fn write_raw<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        let limbs: [u64; 4] = unsafe { std::mem::transmute_copy(&self) };
+        for limb in limbs {
+            writer.write_all(&limb.to_le_bytes())?;
+        }
+        Ok(())
+    }
+}
+
+struct AccessiblePallasPoint {
+    pub x: Fp,
+    pub y: Fp,
+    pub z: Fp,
+}
+
+impl From<pallas::Point> for AccessiblePallasPoint {
+    fn from(p: pallas::Point) -> AccessiblePallasPoint {
+        let point = unsafe { std::mem::transmute::<pallas::Point, AccessiblePallasPoint>(p) };
+        point
+    }
+}
+
+impl From<AccessiblePallasPoint> for pallas::Point {
+    fn from(ap: AccessiblePallasPoint) -> pallas::Point {
+        let p = unsafe { std::mem::transmute::<AccessiblePallasPoint, pallas::Point>(ap) };
+        p
+    }
+}
+
+impl SerdeObject for pallas::Point {
+    fn from_raw_bytes_unchecked(bytes: &[u8]) -> Self {
+        debug_assert_eq!(bytes.len(), 3 * FP_SIZE);
+        let [x, y, z] =
+            [0, 1, 2].map(|i| Fp::from_raw_bytes_unchecked(&bytes[i * FP_SIZE..(i + 1) * FP_SIZE]));
+
+        Self::from(AccessiblePallasPoint { x, y, z })
+    }
+    fn from_raw_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 3 * FP_SIZE {
+            return None;
+        }
+        let [x, y, z] =
+            [0, 1, 2].map(|i| Fp::from_raw_bytes(&bytes[i * FP_SIZE..(i + 1) * FP_SIZE]));
+        x.zip(y).zip(z).and_then(|((x, y), z)| {
+            let res = Self::from(AccessiblePallasPoint { x, y, z });
+            // Check that the point is on the curve.
+            bool::from(res.is_on_curve()).then(|| res)
+        })
+    }
+    fn to_raw_bytes(&self) -> Vec<u8> {
+        let mut res = Vec::with_capacity(3 * FP_SIZE);
+        Self::write_raw(self, &mut res).unwrap();
+        res
+    }
+    fn read_raw_unchecked<R: std::io::Read>(reader: &mut R) -> Self {
+        let [x, y, z] = [(); 3].map(|_| Fp::read_raw_unchecked(reader));
+
+        Self::from(AccessiblePallasPoint { x, y, z })
+    }
+    fn read_raw<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let x = Fp::read_raw(reader)?;
+        let y = Fp::read_raw(reader)?;
+        let z = Fp::read_raw(reader)?;
+        Ok(Self::from(AccessiblePallasPoint { x, y, z }))
+    }
+    fn write_raw<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        let p = AccessiblePallasPoint::from(*self);
+        p.x.write_raw(writer)?;
+        p.y.write_raw(writer)?;
+        p.z.write_raw(writer)
+    }
+}
+
+struct AccessibleVestaPoint {
```

Comment:
This one has the same code as `AccessiblePallasPoint`, the only difference is the call to `.is_on_curve()`.  So I also wonder if the code can be deduplicated :thinking: